promql/engine.go

```diff
@@ -1464,12 +1464,12 @@ func intersection(ls1, ls2 labels.Labels) labels.Labels {
 }
 
 type groupedAggregation struct {
-	labels           labels.Labels
-	value            float64
-	valuesSquaredSum float64
-	groupCount       int
-	heap             vectorByValueHeap
-	reverseHeap      vectorByReverseValueHeap
+	labels       labels.Labels
+	value        float64
+	varianceMean float64
```

Comment:
We're now calculating the mean and the mean for the stddev differently, which is a bit weird. Can you converge them?

Reply:
When you say converge them, would you like to see `varianceMean` renamed to `mean` and used directly for `itemAvg`? I suspect that will be less efficient than the current implementation as it will involve a division operation for each `s` rather than just once per `groupingKey`.  We require the intermediate mean values in `itemStddev` and `itemStdvar` which is why we perform the calculation that way, but it's not needed for `itemAvg`.

Reply:
> When you say converge them, would you like to see varianceMean renamed to mean and used directly for itemAvg? 

That's what I was thinking. There's existing benchmarks if you want to test it out, but I don't imagine it'll make much of a difference as I can't imagine more than 10k items being averaged at once.

Reply:
Do you also want to update `funcAvgOverTime` to make that consistent too?

It seems a little strange to introduce a (minor) performance regression (and imho make it more difficult for someone reading the code to understand it) just to use the same method of calculating the mean when we don't actually need the intermediate values.

Reply:
I'm okay with leaving that one, it's not all in the same function.

This method of calculating mean is also more resilient to floating point issues.